integration_test/update_test.go

```diff
@@ -34,3 +47,62 @@ func TestKrewUpdate(t *testing.T) {
 		t.Fatal("re-run of 'update' must succeed")
 	}
 }
+
+func TestKrewUpdateListsNewPlugins(t *testing.T) {
+	skipShort(t)
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test = test.WithIndex()
+	if err := os.Remove(filepath.Join(test.Root(), "index", "plugins", validPlugin+constants.ManifestExtension)); err != nil {
+		t.Fatalf("failed to delete manifest of an existing plugin: %v", err)
+	}
+
+	out := string(test.Krew("update").RunOrFailOutput())
+	if !strings.Contains(out, "New plugins available:") {
+		t.Fatalf("output doesn't list new plugins available; output=%s", out)
+	}
+	if !strings.Contains(out, validPlugin) {
+		t.Fatalf("output doesn't list the new plugin (%s) is available: %s", validPlugin, out)
+	}
+}
+
+func TestKrewUpdateListsUpgradesAvailable(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+	test = test.WithIndex()
+
+	// set version of some manifests to v0.0.0
+	modifyManifestVersion(t, filepath.Join(test.Root(), "index", "plugins", validPlugin+constants.ManifestExtension), "v0.0.0")
+	modifyManifestVersion(t, filepath.Join(test.Root(), "index", "plugins", validPlugin2+constants.ManifestExtension), "v0.0.0")
```

Comment:
I don't quite understand why you use two plugins here. Is it just to ensure that the output lists all available upgrades? For me that's ok, but not necessary.

Reply:
honestly no need, but also no harm.